update.sh

```diff
@@ -111,65 +204,108 @@ rm -f etc/maglev.demo.key
-            exit 0
-        fi
-    fi
-    # create a clean slate
+[[ "$MAGLEV_SOURCE" == "$MAGLEV_HOME" ]] || {
+  echo "[Info] Coping MagLev to target location $MAGLEV_HOME"
+  if [[ -d "$MAGLEV_HOME" ]]
+  then rm -rf "$MAGLEV_HOME"
+  fi
+  cp -rf "$MAGLEV_SOURCE" "$MAGLEV_HOME"
+  # Make sure we are in MagLev target directory
+  builtin cd "${MAGLEV_SOURCE}"
+}
 
-    if [ -e etc/conf.d/maglev.conf ]; then
-        echo "[Info] In-place upgrade not possible. Removing existing 'maglev' configuration file."
-        rake stone:destroy[maglev] >/dev/null
-    fi
+# setup topaz environment
+export GEMSTONE GEMSTONE_GLOBAL_DIR GEMSTONE_SYS_CONF GEMSTONE_DATADIR GEMSTONE_LOG
+GEMSTONE="${MAGLEV_HOME}/gemstone"
+GEMSTONE_GLOBAL_DIR="$MAGLEV_HOME"
+GEMSTONE_SYS_CONF="${MAGLEV_HOME}/etc/system.conf"
+GEMSTONE_DATADIR="${MAGLEV_HOME}/data/${STONENAME}"
+GEMSTONE_LOG="${MAGLEV_HOME}/log/${STONENAME}/${STONENAME}.log"
 
-    rake build:clobber
-    extent0='gemstone/bin/extent0.dbf'
-    echo "[Info] Building new extent0.ruby.dbf from $extent0 and creating default maglev stone"
-    echo "This could take a while..."
-    if [ -e $extent0 ]; then
-        # NOTE: build:maglev will also create the maglev stone
-        if rake build:maglev ; then
-            echo "[Info] Generating the MagLev HTML documentation"
-            rake rdoc >/dev/null 2>&1
-        else
-            echo "[Error] Could not build new ruby extent. This means there was an error loading the Smalltalk code."
-            exit 1
-        fi
-    else
-        echo "[Error] Can't find ${extent0}: Skip building ruby extent. This means your GemStone download is broken."
-        exit 1
-    fi
-    echo "[Info] Starting MagLev stone (loading kernel classes)"
-    if rake maglev:start ; then
-        echo "[Info] Successfully loaded kernel classes"
+# Create a default repository called "maglev" and generate the MagLev HTML documentation
+#TODO: stop stone if running
+
+# create a clean slate
+if
+  [[ -e etc/conf.d/maglev.conf ]]
+then
+  echo "[Info] Removing existing '${STONENAME}' configuration file."
+  rm -rf "${GEMSTONE_DATADIR}"
+  rm -rf "${MAGLEV_HOME}/etc/conf.d/${STONENAME}.conf"
+  rm -rf "${MAGLEV_HOME}/log/${STONENAME}"
+fi
+if
+  [[ -e "${MAGLEV_HOME}/bin/extent0.ruby.dbf" ]]
+then
+  [[ -e etc/conf.d/maglev.conf ]] || build_maglev_stone_create
+else
+  extent0='gemstone/bin/extent0.dbf'
+  if
+    [[ -e $extent0 ]]
+  then
+    if
+      build_maglev
+    then
+      if
+        [[ $DISABLE_INSTALL_DOC == 0 ]] &&
+        [[ -x "${MAGLEV_HOME}/bin/rake" ]]
+      then
+        echo "[Info] Generating the MagLev HTML documentation"
+        env GEM_HOME="" GEM_PATH="${MAGLEV_HOME}/lib/maglev/gems/1.8/gems" \
+          "${MAGLEV_HOME}/bin/rake" rdoc >/dev/null 2>&1
+      fi
     else
-        echo "[Error] Failed loading kernel classes!"
-        exit 1
+      echo "[Warning] Could not build new ruby extent"
```

Comment:
Will MagLev run w/o this extent? If not, it should be a fatal error and we should `exit`.